### ISSUE TITLE
Xcode 8 Cannot start tests

### ISSUE THREAD

**Issue by benasher44** [travis-ci/travis-ci#6422]
I'm using `xcodebuild` to run tests in Xcode 8 because using `xctool` to run tests under Xcode 8 doesn't appear to work. Now that I've made it through the code signing issues and issues with build output getting cut off with trying to get tests to run using plain `xcodebuild`, the last issue I have (I think) is that every build errors just after code signing (the next step is to actually run the test suite). If I scroll up from the bottom to see if anything obvious went wrong, I consistently get this output:

> 2016-08-08 03:05:35.644 xcodebuild[21425:26294] Error Domain=IDETestOperationsObserverErrorDomain Code=3 "Canceling tests due to timeout in Checking test manager availability... If you believe this error represents a bug, please attach the log file at /Users/travis/Library/Developer/Xcode/DerivedData/ShapeSwift-fiqngwhcdjgmzxcsctffxqjeebgz/Logs/Test/BD54C6B2-AEBE-4E65-B478-6F3F8B5A0138/Session-ShapeSwiftTests-2016-08-08_030332-Ot8pM4.log" UserInfo={NSLocalizedDescription=Canceling tests due to timeout in Checking test manager availability... If you believe this error represents a bug, please attach the log file at /Users/travis/Library/Developer/Xcode/DerivedData/ShapeSwift-fiqngwhcdjgmzxcsctffxqjeebgz/Logs/Test/BD54C6B2-AEBE-4E65-B478-6F3F8B5A0138/Session-ShapeSwiftTests-2016-08-08_030332-Ot8pM4.log}
>
>
>
>2016-08-08 03:05:35.761 xcodebuild[21425:28004] Connection peer refused channel request for "dtxproxy:XCTestManager_IDEInterface:XCTestManager_DaemonConnectionInterface"; channel canceled <DTXChannel: 0x7f99a3baaab0>

Based on this, it appears that an important xpc connection is dying midway through the tests. Then when it's done with all of the building and code signing, running the tests fails and exits silently (no specific error messages other than Xcode thinking that the tests failed, even though they never ran). Running the command in question works fine and runs the tests as expected locally. Sample build:

https://travis-ci.org/benasher44/shapeswift/builds/150521198

That is one case, but it reproduces each time. The only reason I think that this isn't a dup #4725 is because this is for Xcode 8, and it happens every single time (not intermittent).

**Comment by mxcl** [comment]
Same:

* https://travis-ci.org/PromiseKit/AssetsLibrary/builds/152277145

**Comment by pcantrell** [comment]
And with Siesta: https://travis-ci.org/bustoutsolutions/siesta/jobs/154629440

In that job, note that the failure happens before it’s even starting _building_ the test target, much less compiling it.

**Comment by lennartstolz** [comment]
We see this issue on one of our projects as well. 

To ensure it's unrelated to the project's specific setup I've created an empty project but the issue still occurs:

````console
Error Domain=IDETestOperationsObserverErrorDomain Code=3 "The operation couldn’t be completed. (DTXProxyChannel error 1.)
````

See the following PR: [#1 @ travis-ci-xcode8](https://github.com/lennartstolz/travis-ci-xcode-8/pull/1)

Perhaps it will help to solve the issue. 😸

**Comment by benasher44** [comment]
This is still happening in Xcode 8 beta 6: https://travis-ci.org/benasher44/shapeswift/builds/154769480

**Comment by pcantrell** [comment]
Still happening with Xcode 8 GM: https://travis-ci.org/bustoutsolutions/siesta/jobs/158361354#L792

Version number that shows GM here: https://travis-ci.org/bustoutsolutions/siesta/jobs/158361354#L131

**Comment by benasher44** [comment]
Looks like the workaround in #6251 works: setting the destination flag with the iPhone SE simulator. It'd be nice to have a better error message though.

**Comment by pcantrell** [comment]
Some builds that were failing now appear to succeed. However, it’s still consistently failing with iOS 8.1 / iPhone 4s. [Here’s an example.](https://travis-ci.org/bustoutsolutions/siesta/jobs/158652426)

The workaround @benasher44 mentioned doesn’t work for me. Many of my previously failing builds were already using the iPhone SE. Those same builds now succeed when I rerun the old job. It’s only the 8.1/4s combo that’s failing in my suite.

**Comment by benasher44** [comment]
Ah. I should note the workaround I used was specifically for iOS 10

**Comment by mxcl** [comment]
Can confirm, has fixed it for me. I was very dubious that just setting the simulator to SE rather than what I was using (6S) would fix it, but it does. Was getting consistent fails (never succeeded), but am now getting consistent successes (never fails).

**Comment by pcantrell** [comment]
iOS 8.1 problems I mentioned above were my own fault, not Travis’s.

It looks like this is fixed. Builds that were failing before are now working with no changes to the project — on all devices, not just SE.

**Comment by mxcl** [comment]
Just to clarify, nothing was fixed for me with Xcode 8 GM until I switched from specifying 6S to SE. I was specifying by UUID in both cases. *AND* waiting for the simulator to boot before running the tests.

**Comment by pcantrell** [comment]
It was still broken with 8 GM for me too, but spontaneously fixed itself quite recently. @mxcl, try going back to an old build that failed and manually rerunning it.

**Comment by benasher44** [comment]
Going to close this, since it does work if you set your destination to iPhone SE

**Comment by downie** [comment]
@benasher44 I don't think that workaround is 100% reliable. I've got [this project](https://travis-ci.org/blockchain-certificates/cert-wallet) failing travis-ci tests with the same `Domain=IDETestOperationsObserverErrorDomain Code=3` error.

**Comment by jberkel** [comment]
it's definitely not reliable. also what happens if you need to run tests on iPad?

**Comment by garethcurtis** [comment]
Hi. Just had the same issue. I used the iPhone SE workaround yesterday and it was fine but today, now failing again! No changes made but fails with the same error message (with iPhone SE still too). Pretty weird.

2016-09-20 06:59:57.650 xcodebuild[1499:4631] Error Domain=IDETestOperationsObserverErrorDomain Code=3 "The operation couldn’t be completed. (DTXProxyChannel error 1.) If you believe this error represents a bug, please attach the log file at /Users/travis/Library/Developer/Xcode/DerivedData/Forevr-dbrukywaivxzadhasejwvrrogrsq/Logs/Test/29D4C835-2794-4241-9FFE-0C196D982B36/Session-ForevrUITests-2016-09-20_065549-tBurTC.log" UserInfo={NSLocalizedDescription=The operation couldn’t be completed. (DTXProxyChannel error 1.) If you believe this error represents a bug, please attach the log file at /Users/travis/Library/Developer/Xcode/DerivedData/Forevr-dbrukywaivxzadhasejwvrrogrsq/Logs/Test/29D4C835-2794-4241-9FFE-0C196D982B36/Session-ForevrUITests-2016-09-20_065549-tBurTC.log}
** TEST FAILED **

**Comment by 0xced** [comment]
I also just got this error with the Xcode 7.0.1 image, so it seems it’s not an Xcode 8 only issue. https://travis-ci.org/AFNetworking/AFNetworking/jobs/161307588#L2953

**Comment by mvandervelden** [comment]
Also for me the iPhoneSE workaround stopped working (when I checked yesterday). Seems like the simulators get corrupted or something?

2016-09-22 11:24:13.850 xcodebuild[1684:5269] Error Domain=IDETestOperationsObserverErrorDomain Code=3 "The operation couldn’t be completed. (DTXProxyChannel error 1.) If you believe this error represents a bug, please attach the log file at /...}

**Comment by nasli** [comment]
Could be related https://github.com/travis-ci/travis-ci/issues/6643

**Comment by NachoSoto** [comment]
Any update on this? It's making Travis basically useless for iOS tests 😕

**Comment by nasli** [comment]
I have travis and Xcode running with iOS test. Just use xcodebuild with xcpretty output

**Comment by michaelkirk** [comment]
Based on the different error messages, I think @nasli might have been having a separate issues.

**Comment by piv199** [comment]
iPhone SE minimizes errors, but they are still here and unpredictable

**Comment by pcantrell** [comment]
Yeah, I'm consistently seeing about 25% spurious CI failure rate for Xcode builds — much worse during US business hours, much better late night & weekend. My working theory is that it's just due to Travis being so slow: the build times out waiting for the iOS simulator to start, and the SE starts slightly faster and is thus slightly less likely to trip the timeout.

**Comment by pcantrell** [comment]
I managed to wring some more diagnostics out of the build, dumping that log file that the error message tells us to look at. I don't know if there's useful information in it, but here's the log trimmed to the interesting bits, with redundant messages suppressed:

```
(null) xcodebuild[1520:4651] Launching with Xcode.IDEFoundation.Launcher.PosixSpawn
20:51:57.779 xcodebuild[1520:4651] Initializing test infrastructure.
20:51:57.796 xcodebuild[1520:4656] 1A6EAD45-0A28-421A-A6D4-AD0866464FAA: Registered for sim device notification, got token 4
20:51:57.796 xcodebuild[1520:4656] Sim iPhone SE is not booted, will wait up to 120 seconds.
20:51:58.156 xcodebuild[1520:4651] Received Ready from iOSSimulator
20:51:58.157 xcodebuild[1520:4651] Sending notification: NSConcreteNotification 0x7fa5545db860 {name = com.apple.iphonesimulator.startSession; userInfo = {
    deviceUDID = "1A6EAD45-0A28-421A-A6D4-AD0866464FAA";
}}
20:51:58.232 xcodebuild[1520:4684] 1A6EAD45-0A28-421A-A6D4-AD0866464FAA: Received sim device notification: {
    device = "SimDevice : iPhone SE (1A6EAD45-0A28-421A-A6D4-AD0866464FAA) : state={ Booting } deviceType={ SimDeviceType : com.apple.CoreSimulator.SimDeviceType.iPhone-SE } runtime={ SimRuntime : 10.0 (14A345) - com.apple.CoreSimulator.SimRuntime.iOS-10-0 }";
    "new_state" = 2;
    notification = "device_state";
    "prev_state" = 1;
}

<< message repeated 3 times >>

20:51:58.233 xcodebuild[1520:4684] Sim iPhone SE state changed to 2
20:51:58.233 xcodebuild[1520:4686] /Users/travis/Library/Developer/CoreSimulator/Devices: Received sim device set notification: {
    device = "SimDevice : iPhone SE (1A6EAD45-0A28-421A-A6D4-AD0866464FAA) : state={ Booting } deviceType={ SimDeviceType : com.apple.CoreSimulator.SimDeviceType.iPhone-SE } runtime={ SimRuntime : 10.0 (14A345) - com.apple.CoreSimulator.SimRuntime.iOS-10-0 }";
    "new_state" = 2;
    notification = "device_state";
    "prev_state" = 1;
}

<< message repeated 109 times >>

20:51:59.553 xcodebuild[1520:4684] Sim iPhone SE state changed to 3
20:51:59.553 xcodebuild[1520:4684] /Users/travis/Library/Developer/CoreSimulator/Devices: Received sim device set notification: {
    device = "SimDevice : iPhone SE (1A6EAD45-0A28-421A-A6D4-AD0866464FAA) : state={ Booted } deviceType={ SimDeviceType : com.apple.CoreSimulator.SimDeviceType.iPhone-SE } runtime={ SimRuntime : 10.0 (14A345) - com.apple.CoreSimulator.SimRuntime.iOS-10-0 }";
    "new_state" = 3;
    notification = "device_state";
    "prev_state" = 2;
}

<< message repeated 95 times >>

20:51:59.567 xcodebuild[1520:4656] 1A6EAD45-0A28-421A-A6D4-AD0866464FAA: Unregistering for sim device notification with token 4
20:51:59.568 xcodebuild[1520:4656] Sim iPhone SE is booted (state = 3)
20:51:59.568 xcodebuild[1520:4656] Calling -[SimDevice getenv:error:] for TESTMANAGERD_SIM_SOCK
20:51:59.569 xcodebuild[1520:4907] /Users/travis/Library/Developer/CoreSimulator/Devices: Received sim device set notification: {
    device = "SimDevice : iPhone SE (1A6EAD45-0A28-421A-A6D4-AD0866464FAA) : state={ Booted } deviceType={ SimDeviceType : com.apple.CoreSimulator.SimDeviceType.iPhone-SE } runtime={ SimRuntime : 10.0 (14A345) - com.apple.CoreSimulator.SimRuntime.iOS-10-0 }";
    "new_state" = 3;
    notification = "device_state";
    "prev_state" = 2;
}

<< message repeated 11 times >>

20:51:59.775 xcodebuild[1520:4656] Returning /private/tmp/com.apple.launchd.BY5JqiI2FP/com.apple.testmanagerd.unix-domain.socket for TESTMANAGERD_SIM_SOCK
20:51:59.775 xcodebuild[1520:4656] Sim iPhone SE has testmanagerd socket at /private/tmp/com.apple.launchd.BY5JqiI2FP/com.apple.testmanagerd.unix-domain.socket
20:51:59.775 xcodebuild[1520:4656] Connected socket 34 to testmanagerd for Sim iPhone SE
20:51:59.983 xcodebuild[1520:4656] Creating the test bundle connection.
20:52:00.122 xcodebuild[1520:4656] Listening for proxy connection request from the test bundle (all platforms)
20:52:00.122 xcodebuild[1520:4656] Resuming the connection.
20:52:00.122 xcodebuild[1520:4656] Test connection requires daemon assistance.
20:52:00.123 xcodebuild[1520:4651] Starting test session with ID <__NSConcreteUUID 0x7fa554a0bf50> 18FDD69E-38ED-43AE-A66B-9CA492A4FFED
20:52:00.123 xcodebuild[1520:4651] Checking test manager availability..., will wait up to 120s
20:53:38.864 xcodebuild[1520:6879] Closing socket 34 to testmanagerd for Sim iPhone SE
20:53:38.883 xcodebuild[1520:6880] Error from testmanagerd: The operation couldn’t be completed. (DTXProxyChannel error 1.) ((null))
20:53:38.884 xcodebuild[1520:4651] Test operation failure: The operation couldn’t be completed. (DTXProxyChannel error 1.)
20:53:38.885 xcodebuild[1520:4651] _finishWithError:Error Domain=IDETestOperationsObserverErrorDomain Code=3 "The operation couldn’t be completed. (DTXProxyChannel error 1.)" UserInfo={NSLocalizedDescription=The operation couldn’t be completed. (DTXProxyChannel error 1.)} didCancel: 1
```

Complete build is here: https://travis-ci.org/bustoutsolutions/siesta/jobs/167946526

And for comparison, here's a build that didn't blow up: https://travis-ci.org/bustoutsolutions/siesta/jobs/167946527

It seems to be “Checking test manager availability” that times out. In the broken build, it gives up after 98 seconds (despite saying that it will wait for 120…?!?). In the build that works, it succeeds after 72. That certainly suggests things are operating in a success window narrow enough that the ebb and flow of Travis performance could tip it one way or the other.

**Comment by mc7h** [comment]
I don't know if this will help, but I was experiencing the *exact* same problem with another popular CI server. The issue was caused by an overzealous plug-in which was killing "old" processes mid-build.

**Comment by atsakiridis** [comment]
Getting bitten by this as well. Tried:
- Changing target simulator to 'iPhone SE'
- Explicitly setting iOS version to 10.1
- Checking the Build Phases
But none of them worked. This essentially renders all iOS testing unusable in Travis :(. Does anybody know of any workarounds?

Best regards

**Comment by victorg1991** [comment]
@atsakiridis I use travis_retry xcodebuild...

Then if xcodebuild fails it will retry 3 times..

**Comment by atsakiridis** [comment]
Thanks a lot @victorg1991, indeed using `travis_retry` seems to work most of the times. 

Here's the successful build command I used in case it helps anyone:
```
set -o pipefail && travis_retry xcodebuild test -workspace Test-App/Sample.xcworkspace -scheme Sample -destination 'platform=iOS Simulator,name=iPhone SE,OS=10.0' | xcpretty
```

Looking forward to this being fixed though, since sporadic failures are likely to occur, as the timeout is still there :(

**Comment by rmnblm** [comment]
I also have this problem, see https://travis-ci.org/rmnblm/Raclette/builds/176668440 for reference. I'm going to use the proposed workaround from @victorg1991 but I hope this will be fixed in the near future. 🙄

**Comment by justbaum30** [comment]
I have started seeing this issue significantly more since we starting to use the Xcode8.1 images. Does anyone know if `travis_retry` will retry from a simple unit test failure?

**Comment by rmnblm** [comment]
I don't think it will retry when a simple unit test fails. As stated in the documentation, it retries when a command returns non-zero.

> For commands which do not have a built in retry feature, use the `travis_retry` function to retry it up three times if the return code is non-zero

**Comment by cotsog** [comment]
Hey all,

Can you try warming up/pre-starting a particular iOS simulator like this and let us know if it helps mitigate the issue you are seeing?

```
before_install:
  - export IOS_SIMULATOR_UDID=`instruments -s devices | grep "iPhone 6 (10.0" | awk -F '[ ]' '{print $4}' | awk -F '[\[]' '{print $2}' | sed 's/.$//'`
  - echo $IOS_SIMULATOR_UDID
  - open -a "simulator" --args -CurrentDeviceUDID $IOS_SIMULATOR_UDID
```

It's also possible to hardcode the sim UDID but the above recipe makes it compatible with image updates.

Thank you for keeping us posted on your results.

**Comment by michaelkirk** [comment]
> Can you try warming up/pre-starting a particular iOS simulator like this and let us know if it helps mitigate the issue you are seeing?

I made this change previously. Starting a simulator at the beginning of my build helped, but did not solve the issue. I'd say roughly that it moved failure from 75% to 25% of the time.

**Comment by heshamsalman** [comment]
I'm still running into a failure on my end.

**Comment by cotsog** [comment]
@michaelkirk: are you using `travis_retry` in front of your `xcodebuild` command as well?

@heshamsalman: can you send a URL/link of to a failing build to support [at] travis-ci [dot] com?

**Comment by heshamsalman** [comment]
@cotsog adding iPhone SE _in addition to_ the retry seemed to have fixed it 👍

**Comment by justbaum30** [comment]
@cotsog Launching the simulator beforehand did not fix the issue. 

I'm currently testing retry, but I don't find this solution to be a real long-term solution considering my pull request has a nearly 100% failure rate while all the other pull requests on the project has 100% success rate. This will add even more to the build times to the already slow Travis build times. 

The only difference is that my pull request is using the Xcode8.1 on Sierra image while the rest are on Xcode8.0. I've also noticed that build time is almost twice as long. I think there are some issues with the new Xcode image that needs to be looked at/worked out.

**Comment by brianmichel** [comment]
> The only difference is that my pull request is using the Xcode8.1 on Sierra image while the rest are on Xcode8.0. I've also noticed that build times are also almost twice as long. I think there are some issues with the new Xcode image that needs to be looked at/worked out.

@cotsog we are experiencing similar timing and failures as outlined here, we recently switched to the 8.1 image as well.

**Comment by pcantrell** [comment]
First experiments suggest that warm up + retry has solved the issue for me — and also nearly doubled build times. I did this slightly tweaked version of what @cotsog suggested:

```yaml
env:
  - scheme='Siesta macOS' platform='OS X'
  - scheme='Siesta iOS'   platform='iOS Simulator' ios_version='10.1' ios_device='iPhone SE'
  - scheme='Siesta iOS'   platform='iOS Simulator' ios_version='9.3'  ios_device='iPhone 6'
  - scheme='Siesta iOS'   platform='iOS Simulator' ios_version='8.1'  ios_device='iPhone 4s'

before_install: |
  ios_simulator_udid=$(instruments -s devices | grep "$ios_device ($ios_version)" | sed -E 's/.*\[([0-9A-F-]+)\].*/\1/g')
  echo "Prelaunching iOS simulator with UDID $ios_simulator_udid"
  open -a "simulator" --args -CurrentDeviceUDID $ios_simulator_udid
```

…plus `travis_retry xcodebuild …` per @victorg1991.

**Comment by elorz007** [comment]
A bit far fetched but I will post my solution just in case. I had the same `IDETestOperationsObserverErrorDomain Code=3`. For me the issue appeared when I moved to Xcode 8 and from cocoapods 0.39.0 to 1.1.1. After many attempts with warming-up the simulators, etc. what fixed it for me was adding the `inherit! :search_paths` in my Podfile. 

```
target :MyAppTests do
  inherit! :search_paths
  [...]
end
```
More info: http://blog.cocoapods.org/CocoaPods-1.0-Migration-Guide/

**Comment by mc7h** [comment]
We were experiencing issues with our CI system. These problems were resolved by allocating more memory to the build agents.

**Comment by elorz007** [comment]
It looks like most of the instabilities with launching simulators are caused by a bug in Xcode 8.1 that is supposedly addressed in Xcode 8.2 beta

**Comment by werner77** [comment]
See my solution here: http://stackoverflow.com/a/40790171/480467

The problem lies in the fact that the timeout counter in xcodebuild starts as soon as the command is executed, so the compile time counts towards the timeout (which is 120 seconds)

**Comment by atsakiridis** [comment]
Very interesting, so the reason this didn't happen in previous version is that the timeout was different or the simulator took longer to come up I guess?

**Comment by janderson-primary** [comment]
@werner77 thank you!  your fix appears to have solved this problem for us.

**Comment by shorbenko** [comment]
Submitted bug report to Apple. 
I don't use Travis but have the same issue with iOS Simulator and XCode 8.3.2

**Comment by gparente** [comment]
@shorbenko I see these issue on XCode 8.3.1 as well. In my case the error actually happens _in-between_ tests causing some of them to succeed and some of them to fail. Anyone else seeing something similar?

**Comment by shorbenko** [comment]
@gparente No, I cannot start tests at all.

`05:58:15.127 xcodebuild[64907:450465] _finishWithError:Error Domain=IDETestOperationsObserverErrorDomain Code=3 "The operation couldn’t be completed. (DTXProxyChannel error 1.)" UserInfo={NSLocalizedDescription=The operation couldn’t be completed. (DTXProxyChannel error 1.)} didCancel: 1
`

**Comment by TheSkwiggs** [comment]
I started having this today exactly too, none of my unit tests run anymore :(

**Comment by asalom** [comment]
I had this once today too, re-executed the job and the error was gone: ```iOSSimulator: Could not launch simulator: -1712```
